File: python/443.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        last = chars[0]
        count = 0
        res = chars[0]
        for c in chars:
            if(c == last):
                count+=1
            else:
                if count>1:
                    res+=str(count)
                last = c
                count = 1
                res+=c
        if count>1:
            res+=str(count)
        
        chars.clear()
        for i in range(len(res)):
            chars.append(res[i])

        print(chars)
        return len(chars)

        

File: cpp/443.cpp
class Solution {
public:
    int compress(vector<char>& chars) {
        string res = "";
        res+=chars[0];
        char last = chars[0];
        int count = 1;
        for(int i=1; i<chars.size(); i++) {
            if(chars[i]==res.back()) 
                count++;
            else {
                if(count>1)
                    res+=to_string(count);
                res+=chars[i];
                count =1;
            }
        }
        if(count>1)
                res+=to_string(count);
        
        chars.clear();
        for(auto& c : res)
            chars.push_back(c);

        return res.size();
    }
};